project/spelling/misspell.py

```diff
@@ -0,0 +1,212 @@
+    letters = grapheme.letters
+    regex = re.compile(".*?([aeioruy]+)$")
+    result = regex.match(letters)
+
+    if result is None:
+        return grapheme, None
+    else:
+        vowels = result.group(1)
+        carry = letters[:len(letters) - len(vowels)]
+        return Grapheme(carry), Grapheme(vowels)
+
+
+def find_consonant(letters, phoneme):
+    for i in range(len(letters) + 1, 1, -1):
+        substring = letters[:i]
+
+        if substring in consonants.search[phoneme]:
+            return substring
+
+
+def graphemes(word):
+    """
+    Converts a word into its *graphemes* - i.e. separating
+    the words into letters, with each chunk of letters representing
+    either a consonant, a vowel or a vowel cluster.
+    """
+    final = []
+    word_phonemes = phonemes(word)
+
+    # Iterate over every phoneme in a word
+    for phoneme in word_phonemes:
+        # If that phoneme is a consonant:
+        if phoneme in consonants.search.keys():
+            # Find the first grapheme that matches that phoneme,
+            # and include the extra letters behind it
+            extra, grapheme = find_grapheme(word, phoneme)
+
+            crop_length = 0
+
+            # Algorithm to "carry" silent/vowels to consonants to last
+            # consonant, since the consonant algorithm is lazy and gets the
+            # shortest string possible.
+            if len(final) > 0:
+                consonant = final[-1]
+
+                # Creating a "letter pool" of possible characters that could
+                # be in the previous vowel
+                letter_pool = consonant.letters + extra.letters
+                new_consonant = find_consonant(letter_pool, consonant.phoneme)
+
+                # If we *do* find a new consonant:
+                if new_consonant is not None:
+                    # The amount of letters copied over from the extra letters
+                    # to the new consonant
+                    letters_from_extra = (
+                        len(new_consonant) - len(consonant.letters)
+                    )
+                    # We need to delete more letters from "word" since we're
+                    # editing "extra", so add those letters back again
+                    crop_length += letters_from_extra
+                    # Delete carried letters from "extra"
+                    extra.letters = extra.letters[letters_from_extra:]
+                    # Add the carried letters to the previous consonant
+                    consonant.letters = new_consonant
+
+            # If there are still any letters left in the "extra letters",
+            # add it to the end as a vowel cluster
+            if extra.letters != "":
+                final.append(extra)
+
+            final.append(grapheme)
+            crop_length += len(extra.letters) + len(grapheme.letters)
+            # Crop the word since we've already processed some letters
+            word = word[crop_length:]
+        else:
```

Comment:
This is the no-op I was talking about.

Reply:
Yeah, I remembered I had something like that in the spelling code and I just forgot to get rid of it.